src/journal.js

```diff
@@ -0,0 +1,173 @@
+  return res
+}
+
+// create new journal
+const createJournal = (param) => client.tasks.create(param).then((res) => res)
+
+// create sub tasks for each parent task
+const createSubTasks = (taskId, subTasks) => {
+  if (subTasks.length > 0) {
+    subTasks.forEach(async (task) => {
+      // task: { name: '', subTask: [] }
+      const subTaskParam = { name: task.name }
+      if (task.subTasks.length > 0) {
+        await timeout(30 * 1000)
+        client.tasks.addSubtask(taskId, subTaskParam).then((r) => {
+          createSubTasks(r.gid, task.subTasks)
+        })
+      } else {
+        client.tasks.addSubtask(taskId, subTaskParam)
+      }
+    })
+  }
+}
+
+// format completed tasks & subtasks into single string
+const getLastJournalNotes = async (journal) => {
+  const taskArr = journal.completedSubTasks || journal
+  if (taskArr.length > 0) {
+    const notes = taskArr.map(async (j) => {
+      if (j.subTasks.length > 0) {
+        const subNotesString = await getLastJournalNotes(j.subTasks)
+        const subNotes = subNotesString.split('\n')
+        return `${j.name}
+        * ${subNotes.join('\n        * ')}`
+      }
+      return `${j.name}`
+    })
+    const noteArr = await Promise.all(notes).then((r) => r)
+    return `- ${noteArr.join('\n- ')}`
+  }
+  return ''
+}
+
+const createNewJournals = async () => {
+  if (HD.isHoliday(new Date()) || isWeekend) {
+    return
+  }
+
+  try {
+    const params = {
+      project: DEV_JOURNAL,
+      completed_since: 'now',
+      opt_fields: 'completed,projects.name,due_on,name,notes,subtasks,assignee.name,custom_fields',
+    }
+    const journals = await getJournals(params)
+
+    journals.map(async (journal) => {
+      if (journal.incompleteSubTasks.length > 0 || journal.completedSubTasks.length > 0) {
+        const lastJournalNotes = await getLastJournalNotes(journal)
+
+        // const createJournalParams = {
+        //   name: journal.name,
+        //   assignee: journal.assignee.gid,
+        //   completed: false,
+        //   due_on: `${new Date().getFullYear()}-0${new Date().getMonth() + 1}-${new Date().getDate()}`,
+        //   projects: [journal.projects],
+        //   workspace: journal.workspace,
+        //   custom_fields: { [journal.customField]: lastJournalNotes },
+        // }
+
+        const createJournalParams = {
+          name: `TEST-${journal.name}`,
+          completed: false,
+          due_on: `${new Date().getFullYear()}-0${new Date().getMonth() + 1}-${new Date().getDate()}`,
+          projects: [journal.projects],
+          workspace: journal.workspace,
+          custom_fields: { [journal.customField]: lastJournalNotes },
+        }
+
+        await timeout(30 * 1000)
+        const newJournal = await createJournal(createJournalParams)
+        await timeout(60 * 1000)
+        createSubTasks(newJournal.gid, journal.incompleteSubTasks)
+      }
+    })
+  } catch (e) {
+    console.error(e)
+  }
+}
+
+createNewJournals()
```

Comment:
you can incorporate this into `routine.js` which also handles rate limited retries etc.

Reply:
question about the rate limit handler in `routine.js`: if i call this in `routine.js`, every time it retires it will call the function again?